crates/rpc/src/handlers/state.rs

```diff
@@ -1,86 +1,163 @@
+                .ok_or_else(|| error::ErrorNotFound("Justified checkpoint not found"))?;
 
             Ok(Some(justified_checkpoint.root))
         }
         ID::Head | ID::Genesis => {
-            return Err(ApiError::NotFound(format!(
+            return Err(error::ErrorNotFound(format!(
                 "This ID type is currently not supported: {state_id:?}"
             )));
         }
         ID::Slot(slot) => db.slot_index_provider().get(slot),
         ID::Root(root) => db.state_root_index_provider().get(root),
     }
-    .map_err(|_| ApiError::InternalError)?
-    .ok_or(ApiError::NotFound(format!(
-        "Failed to find `block_root` from {state_id:?}"
-    )))?;
+    .map_err(error::ErrorInternalServerError)?
+    .ok_or_else(|| {
+        error::ErrorNotFound(format!("Failed to find `block_root` from {state_id:?}"))
+    })?;
 
     db.beacon_state_provider()
         .get(block_root)
-        .map_err(|_| ApiError::InternalError)?
-        .ok_or(ApiError::NotFound(format!(
-            "Failed to find `beacon_state` from {block_root:?}"
-        )))
+        .map_err(error::ErrorInternalServerError)?
+        .ok_or_else(|| {
+            error::ErrorNotFound(format!("Failed to find `block_root` from {state_id:?}"))
+        })
 }
 
-pub async fn get_state(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}")]
+pub async fn get_beacon_state(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
-    Ok(with_status(BeaconResponse::json(state), StatusCode::OK))
+    Ok(HttpResponse::Ok().json(BeaconResponse::from(state)))
 }
 
-pub async fn get_state_root(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}/root")]
+pub async fn get_state_root(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
     let state_root = state.tree_hash_root();
 
-    Ok(with_status(
-        BeaconResponse::json(RootResponse::new(state_root)),
-        StatusCode::OK,
-    ))
+    Ok(HttpResponse::Ok().json(BeaconResponse::from(state_root.to_string())))
 }
+
+/// Called by `/eth/v1/beacon/states/{state_id}/fork` to get fork of state.
+#[get("/beacon/states/{state_id}/fork")]
+pub async fn get_state_fork(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
+
+    Ok(HttpResponse::Ok().json(state.fork))
```

Comment:
```suggestion
    Ok(HttpResponse::Ok().json(BeaconResponse::new(state.fork)))
```

Can you look through your PR and fix these regressions because, the same regressions show up over and over again